http/client.go

```diff
@@ -27,53 +27,156 @@ func NewClient(c *http.Client, router *mux.Router, endpoint string, t flux.Token
 }
 
 func (c *client) ListServices(_ flux.InstanceID, namespace string) ([]flux.ServiceStatus, error) {
-	return invokeListServices(c.client, c.token, c.router, c.endpoint, namespace)
+	var res []flux.ServiceStatus
+	err := c.get(&res, "ListServices", "namespace", namespace)
+	return res, err
 }
 
 func (c *client) ListImages(_ flux.InstanceID, s flux.ServiceSpec) ([]flux.ImageStatus, error) {
-	return invokeListImages(c.client, c.token, c.router, c.endpoint, s)
+	var res []flux.ImageStatus
+	err := c.get(&res, "ListImages", "service", string(s))
+	return res, err
 }
 
 func (c *client) PostRelease(_ flux.InstanceID, s jobs.ReleaseJobParams) (jobs.JobID, error) {
-	return invokePostRelease(c.client, c.token, c.router, c.endpoint, s)
+	args := []string{"image", string(s.ImageSpec), "kind", string(s.Kind)}
+	for _, spec := range s.ServiceSpecs {
+		args = append(args, "service", string(spec))
+	}
+	for _, ex := range s.Excludes {
+		args = append(args, "exclude", string(ex))
+	}
+
+	u, err := makeURL(c.endpoint, c.router, "PostRelease", args...)
+	if err != nil {
+		return "", errors.Wrap(err, "constructing URL")
+	}
+
+	req, err := http.NewRequest("POST", u.String(), nil)
+	if err != nil {
+		return "", errors.Wrapf(err, "constructing request %s", u)
+	}
+	c.token.Set(req)
+
+	resp, err := executeRequest(c.client, req)
+	if err != nil {
+		return "", errors.Wrap(err, "executing HTTP request")
+	}
+	defer resp.Body.Close()
+
+	var res postReleaseResponse
+	if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
+		return "", errors.Wrap(err, "decoding response from server")
+	}
+	return res.ReleaseID, nil
```

Comment:
Annoyingly, `PostRelease` is the only post with a response body. Would be nice if they could all use the same `c.post` method. Suggestions on how to do that, welcome...